kem/kyber/gen.go

```diff
@@ -7,8 +7,10 @@ package main
 
```

Comment:
Could you please move these files used to generate code to an `internal` folder.
```
/kem/internal/gen.go
/kem/internal/pkg.templ.go
```
Generated packages:
```
/kem/kyber*
/kem/mlkem*
```

Reply:
> Could you please move these files used to generate code to an `internal` folder.

Shall we do that in a separate PR? It's not only Kyber that has a `gen.go` outside of an `internal`.